chapters/functions.tex

```diff
@@ -1567,13 +1567,21 @@ \section{Declaring Inverses of Functions}\label{declaring-inverses-of-functions}
 
 The function in the \lstinline!inverse! annotation must be an actual inverse.
 This requires that $f_2$ is such that, if $u_k$ is calculated as \lstinline!$u_k$ := $f_2$($\ldots$, y, $\ldots$)!, the equality \lstinline!y = $f_1$($\ldots$, $u_k$, $\ldots$)! is satisfied up to a certain precision, for all valid values of the input arguments of \lstinline!$f_2$($\ldots$, y, $\ldots$)!.
-the equality \lstinline!y = $f_1$($\ldots$, $u_k$, $\ldots$)! is satisfied up to a certain precision.
+
+\begin{nonnormative}
+There is no guarantee that a provided inverse will be used, and no rule for at which stage of symbolic processing it could be applied.
+Hence, providing an inverse function cannot be used as a means to select a particular solution to a nonlinear equation with multiple solutions (in other words, to select particular pre-image for a non-injective function).
+On the contrary, providing an inverse function can make a model ambiguous, as it won't be known if an equation is solved using a generic nonlinear equation solver (with solution being influenced by guess values), or solved using the provided inverse (possibly selecting a different solution).
+
+While it is possible to have several inverse functions that could be applied to solve the same equation, the real problem of ambiguity isn't the number of applicable inverse functions, but that there is ambiguity as soon as there is one inverse function and multiple solutions.
```

Comment:
We generally don't use contractions such as "isn't" in the text.

Reply:
Fixed, along with a few more in the new description of figure annotations.  Resolving.